src/client/cli/cmd/transfer.cpp

```diff
@@ -62,6 +70,15 @@ mp::ReturnCode cmd::Transfer::run(mp::ArgParser* parser)
 
             try
             {
+                // NOTE: asked only for streaming, but also has
+                // functionality for pushing and pulling files
+                if (streaming_enabled)
+                {
+                    mp::SFTPClient sftp_client{host, port, username, priv_key_blob};
```

Comment:
Instead of passing in at the `SFTPClient` ctor, I instead just passed in the appropriate `stream` from the `Terminal` member in the call to `stream_file`, which is now an overload based on either `cin` or `cout`.